posts/2025-01-18-Project-Goals-Dec-Update.md

```diff
@@ -0,0 +1,282 @@
+
+Shout-out to [camelid](https://github.com/camelid), [boxy](https://github.com/boxy) and [compiler-errors](https://github.com/compiler-errors).
+
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/104'><strong>Begin resolving &#x60;cargo-semver-checks&#x60; blockers for merging into cargo</strong></a></div>
+    <div style="flex: initial;"><progress value="2" max="4"></progress>
+</div>
+</div>
+
+Over the course of the last six months...
+
+* cargo semver-checks began to include generic parameters and bounds in its schema, allowing for more precise lints;
+* cargo manifest linting was implemented and merged, allowing for lints that look at the cargo manifest;
+* building on cargo manifest linting, the `feature_missing` lint was added, which identifies breakage caused by the removal of a package feature.
+
+In addition, we fleshed out a design sketch for the changes in rustdoc's JSON support that are needed to support cross-crate item linting. This in turn requires compiler extensions to supply that information to rustdoc.
+
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/106'><strong>Const traits</strong></a></div>
+    <div style="flex: initial;"><progress value="0" max="2"></progress>
+</div>
+</div>
+
+* Progress was made on adding const traits and implementation in the compiler, with improvements being carefully considered. `Add` was constified in [rust#133237](https://github.com/rust-lang/rust/pull/133237) and `Deref`/`DerefMut` in [rust#133260](https://github.com/rust-lang/rust/pull/133260).
+* Further progress was made on implementing stability for the const traits feature in [rust#132823](https://github.com/rust-lang/rust/pull/132823) and [rust#133999](https://github.com/rust-lang/rust/pull/133999), with additional PRs constifying more traits open at [rust#133995](https://github.com/rust-lang/rust/pull/133995) and [rust#134628](https://github.com/rust-lang/rust/pull/134628).
+
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/107'><strong>Ergonomic ref-counting</strong></a></div>
+    <div style="flex: initial;"><progress value="1" max="6"></progress>
+</div>
+</div>
+
+* Over the last six months, we created a lang-team experiment devoted to this issue and [spastorino](https://github.com/spastorino) began work on an experimental implementation. [joshtriplett](https://github.com/joshtriplett) authored [RFC 3680](https://github.com/rust-lang/rfcs/3680), which has received substantial feedback. The current work is focused on identifying "cheaply cloneable" types and making it easy to create closures that clone them instead of moving them.
```

Comment:
```suggestion
* Over the last six months, we created a lang-team experiment devoted to this issue and [spastorino](https://github.com/spastorino) began work on an experimental implementation. [joshtriplett](https://github.com/joshtriplett) authored [RFC 3680](https://github.com/rust-lang/rfcs/pull/3680), which has received substantial feedback. The current work is focused on identifying "cheaply cloneable" types and making it easy to create closures that clone them instead of moving them.
```